### ISSUE TITLE
Handling of ECC error

### ISSUE THREAD

**Issue by novykkostia** [dlbeer/dhara#34]
Good day @dlbeer,

While working with your dhara ftl for nand flash it was noticed that there is no ecc error handling, the dhara functions just return this error without any further action. Is it necessary to implement ECC error handling yourself? I saw a similar issue https://github.com/dlbeer/dhara/issues/24

**Comment by dlbeer** [comment]
Yes, it is necessary to do this yourself. Many ECC chips will handle it for you in hardware. If you need a software implementation, there's example code provided for BCH and Hamming codes.

Whichever way you do it, remember to read enough data during a partial read to get a complete ECC block -- partial reads require ECC too.
